silc/force_field/gaff2.py

```diff
@@ -283,10 +283,12 @@ def set_dummy_replacement(self, dummy_replacement):
             raise RuntimeError("Please set dummy replacement first using set_dummy_replacement().")
         if isinstance(dummy_replacement, list) and len(dummy_replacement) == self.num_dummy:
             for i in range(len(dummy_replacement)):
-                if len(dummy_replacement[i]) == 1:
+                if len(dummy_replacement[i]) == 1 or dummy_replacement[i] == "[NH3+]":
                     self.replace_dummy_with.append(dummy_replacement[i])
```

Comment:
Not that important, but we don't really need the `i` here, right? If so, I think you can simplify this a bit, like so:
```suggestion
            for dummy in dummy_replacement:
                if len(dummy) == 1 or dummy == "[NH3+]":
                    self.replace_dummy_with.append(dummy)
```

Reply:
Sure, I will simplify it. Thanks for the suggestion.